WordPress/gutenberg
# Site Title: Add support settings for colors, fonts, and line height

Part of #21087

## Description
Adds block level support for colors, fonts, and line height to the Site Title block.

Adds server side block style support via a function on the `render_block` filter to apply styles set for all blocks supporting `__experimentalColor`, `__experimentalFontSize`, and `__experimentalLineHeight`.

## How has this been tested?
* Local dev environment
* New PHP unit tests

## Testing instructions
* Open the site editor
* Add a site title block if it doesn't already exist
* Modify the alignment, colors, font sizes and line height.
* Add a custom CSS class
* Save, and confirm that all of these changes are represented in the front end.

## TODO

- [x] Support line height and font size styles and classnames on server side
- [x] Adjust the colors array to merge together text and background class and style data
- [x] Add `__experimental*` checks before rendering classnames and styles
- [x] ~Look at supporting `__experimentalPadding` classes/styles~ (This looks too influx to support in this PR).
- [x] Write some PHP unit tests for all the new server side code 

## Screenshots 
<img width="1334" alt="Screen Shot 2020-06-08 at 10 58 54 AM" src="https://user-images.githubusercontent.com/1464705/84064369-56c25480-a977-11ea-9b9d-ee142c51588d.png">

## Types of changes
New feature.

## Checklist:
